### PR TITLE
Replace use of six.reraise

### PR DESCRIPTION
**Pull request by rahulporuri**
## What changes are proposed in this pull request?

Along with #3538 , this PR removes the dependence on six.

This PR specifially handles removing the dependence on `six.reraise`. This was slightly tricky so I chose to simply replace the use of `six.reraise` with the actual implementation of `six.reraise`, removing the unnecessary pieces.

The code used comes from - https://github.com/benjaminp/six/blob/c0be8815d13df45b6ae471c4c436cce8c192245d/six.py#L697-L706

## How is this patch tested?

The changes are slightly non-trivial so this patch was not tested locally. I hope the testsuite will catch any potential issues with this patch.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/mleap.py [hunk 1]
```diff
@@ -267,7 +266,15 @@ def _handle_py4j_error(reraised_error_type, reraised_error_text):
...
-    reraise(reraised_error_type, reraised_error_type(reraised_error_text), tb)
+    exception = reraised_error_type(reraised_error_text)
+
+    try:
+        if exception.__traceback__ is not tb:
+            raise exception.with_traceback(tb)
+        raise exception
+    finally:
+        exception = None
+        tb = None
```
Comment (harupy) [inline mlflow/mleap.py:277]: @rahulporuri What do you think about defining `reraise` somewhere and using it in `mleap.py` and `scoring_server/__init__.py`

Comment (rahulporuri) [reply mlflow/mleap.py:277]: that sounds like a good idea but i won't be able to make that change before the weekend. I hope that's okay.

Comment (harupy) [reply mlflow/mleap.py:277]: That's totally fine :)

**Comment by harupy** [comment]
@rahulporuri Hi are you still working on this?

**Comment by harupy** [comment]
Addressed via #3916

**Comment by rahulporuri** [comment]
Sorry for the radio silence @harupy but i am not able to make time to work on these PRs at the moment. Note that I also have #3539 open - which addresses some more Python 2-specific code.

**Pull request closed without merging**